test/session.test.js

```diff
@@ -0,0 +1,265 @@
+    await sleepUntilSessionExpires(this.application);
+
+    // Performing a sync request with an expired access token.
+    await this.application.sync(syncOptions);
+
+    // After the above sync request is completed, we obtain the session information.
+    const sessionAfterSync = this.application.apiService.getSession();
+
+    expect(sessionBeforeSync).to.not.equal(sessionAfterSync);
+    expect(sessionBeforeSync.accessToken).to.not.equal(sessionAfterSync.accessToken);
+    expect(sessionBeforeSync.refreshToken).to.not.equal(sessionAfterSync.refreshToken);
+    expect(sessionBeforeSync.expireAt).to.be.lessThan(sessionAfterSync.expireAt);
+    // New token should expire in the future.
+    expect(sessionAfterSync.expireAt).to.be.greaterThan(Date.now());
+  }).timeout(20000);
+
+  it('should be consistent between storage and apiService', async function () {
+    const sessionFromStorage = await getSessionFromStorage(this.application);
+    const sessionFromApiService = this.application.apiService.getSession();
+
+    expect(sessionFromStorage).to.equal(sessionFromApiService);
+
+    await this.application.apiService.refreshSession();
+
+    const updatedSessionFromStorage = await getSessionFromStorage(this.application);
+    const updatedSessionFromApiService = this.application.apiService.getSession();
+
+    expect(updatedSessionFromStorage).to.equal(updatedSessionFromApiService);
+  });
+
+  it(' should be performed successfully and terminate session with a valid access token', async function () {
+    const signOutResponse = await this.application.apiService.signOut();
+    expect(signOutResponse.status).to.equal(204);
+
+    const syncResponse = await this.application.apiService.sync([]);
+    expect(syncResponse.status).to.equal(401);
+    expect(syncResponse.error.tag).to.equal('invalid-auth');
+    expect(syncResponse.error.message).to.equal('Invalid login credentials.');
+  }).timeout(20000);
+
+  it('sign out request should be performed successfully and terminate session with expired access token', async function () {
+    // Waiting enough time for the access token to expire, before performing a sign out request.
+    await sleepUntilSessionExpires(this.application);
+
+    const signOutResponse = await this.application.apiService.signOut();
+    expect(signOutResponse.status).to.equal(204);
+
+    const syncResponse = await this.application.apiService.sync([]);
+    expect(syncResponse.status).to.equal(401);
+    expect(syncResponse.error.tag).to.equal('invalid-auth');
+    expect(syncResponse.error.message).to.equal('Invalid login credentials.');
+  }).timeout(20000);
+
+  describe('change password request', async function () {
+    beforeEach(async function () {
+      this.newPassword = Factory.randomString();
+    });
+
+    it('should be successful with a valid access token', async function () {
+      const changePasswordResponse = await this.application.changePassword(
```

Comment:
When running `this.application.changePassword()`, the token will expire before the request is sent to the server (using an expiration time of 2 seconds). A quick workaround would be to use a higher expiration time (but that just increases the overall time the tests are executing). 

I guess this is normal, since a new set of keys are generated.